File: src/main/java/io/vertigo/shiny/components/text/toggle/ShinyToggle.java
package io.vertigo.shiny.components.text.toggle;

import io.vertigo.shiny.components.ShinyComponent;

public record ShinyToggle(
		String label,
		boolean value,
		ShinyToggleType toggleType,
		String onText,
		String offText,
		boolean showText) implements ShinyComponent {

	public String type() {
		return "ShinyToggle";
	}
}

File: banshee/src/models/text/toggle/ShinyToggle.ts
import { ShinyComponent } from '../../../ShinyComponent';
import { ShinyToggleType } from './ShinyToggleType';

export interface ShinyToggle extends ShinyComponent {
  label: string;
  value: boolean;
  toggleType: ShinyToggleType;
  onText?: string;
  offText?: string;
  showText: boolean;
  type: string;
}
